### PR TITLE
feat(TCOMP-2186): discover schema extended

### PR DESCRIPTION
**Pull request by undx**
https://jira.talendforge.org/browse/TCOMP-2186

*Related PRs*
- https://github.com/Talend/component-runtime/pull/641 (master)
- https://github.com/Talend/tdi-studio-se/pull/8072 (master)
- https://github.com/Talend/tdi-studio-se/pull/8095 (maintenance/8.0)

### CODE REVIEW

File: main/plugins/org.talend.sdk.component.studio-integration/components/tTaCoKitGuessSchema/tTaCoKitGuessSchema_begin.javajet [hunk 1]
```diff
@@ -63,24 +63,17 @@ System.setOut(new java.io.PrintStream(new java.io.OutputStream() {
...
+    final String originalName = (String)node.getElementParameter(EParameterName.COMPONENT_NAME.getName()).getValue();
+    final String family = (String)node.getElementParameter(EParameterName.FAMILY.getName()).getValue();
+    final String version = (String)node.getElementParameter(EParameterName.VERSION.getName()).getValue();
 %>
-        <%@include file="../../jet_stub/generic/configuration.javajet"%>
-
-<%  } else { %>
-         // Enforce max batch size to 1 for guess schema
-         globalMap.put("MAX_BATCH_SIZE_<%=guessSchemaCid%>", 1);
-        <%@include file="./tTaCoKitGuessSchema_begin_for_processor.javajet"%>
```
Comment (wwang-talend) [inline main/plugins/org.talend.sdk.component.studio-integration/components/tTaCoKitGuessSchema/tTaCoKitGuessSchema_begin.javajet:82]: need to remove tTaCoKitGuessSchema_begin_for_processor.javajet and so on.

File: main/plugins/org.talend.sdk.component.studio-integration/jet_stub/generic/utils/schema_incoming.javajet [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+           inputConn = conn;
+    	}
+    }
+    List<IMetadataColumn> input_columnList = null;
+    if(inputConn != null) {
+    	input_columnList = inputConn.getMetadataTable().getListColumns();
+    }
+%>
+
+<%@include file="./schema_feed.javajet"%>
```
Comment (wwang-talend) [inline main/plugins/org.talend.sdk.component.studio-integration/jet_stub/generic/utils/schema_incoming.javajet:49]: better to inline this javajet?

File: main/plugins/org.talend.sdk.component.studio-integration/components/tTaCoKitGuessSchema/tTaCoKitGuessSchema_begin.javajet (hunk 1 above)
Comment (undx) [reply main/plugins/org.talend.sdk.component.studio-integration/components/tTaCoKitGuessSchema/tTaCoKitGuessSchema_begin.javajet:82]: yep, it was planned ;-)

File: main/plugins/org.talend.sdk.component.studio-integration/jet_stub/generic/utils/schema_incoming.javajet (hunk 2 above)
Comment (undx) [reply main/plugins/org.talend.sdk.component.studio-integration/jet_stub/generic/utils/schema_incoming.javajet:49]: Done

File: main/plugins/org.talend.sdk.component.studio-integration/components/tTaCoKitGuessSchema/tTaCoKitGuessSchema_begin.javajet [hunk 3]
```diff
@@ -90,13 +83,17 @@ System.setOut(new java.io.PrintStream(new java.io.OutputStream() {
...
-                tacokitGuessSchema.guessInputComponentSchema();
-            } finally {
-                tacokitGuessSchema.close();
-            }
-    <% } else { %>
-        globalMap.put("___tacokitGuessSchema___", tacokitGuessSchema);
-        int _recordsCount_<%=guessSchemaCid%> = 0;
-    <% } %>
\ No newline at end of file
+<% if ("input".equalsIgnoreCase(tacokitComponentType)) {  %>
```
Comment (wwang-talend) [inline main/plugins/org.talend.sdk.component.studio-integration/components/tTaCoKitGuessSchema/tTaCoKitGuessSchema_begin.javajet:86]: seems DiscoverSchemaExtended works for processor and studio, is ok for me now.

**Review by ozhelezniak-talend — APPROVED** [review]
lgtm

**Pull request merged**